### PR TITLE
feat(cli): filter working samples from list

### PR DESCRIPTION
**Pull request by tecton**
Add property `maximumCliVersion` and `minimumCliVersion` for CLI.

Related task: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/25061407

### CODE REVIEW

File: packages/cli/src/utils.ts [hunk 1]
```diff
@@ -200,7 +203,19 @@ export interface Sample {
 
 export async function getTemplates(): Promise<Sample[]> {
```
Comment (wenytang-ms) [inline packages/cli/src/utils.ts:204]: maybe the name should be like getSamples?

**Review by Alive-Fish — APPROVED** [review]

File: packages/cli/src/utils.ts [hunk 2]
```diff
@@ -200,7 +203,19 @@ export interface Sample {
...
   await sampleProvider.fetchSampleConfig();
-  const samples = sampleProvider.SampleCollection.samples.map((sample) => {
+  const version = getVersion();
+  const availableSamples = sampleProvider.SampleCollection.samples.filter(
```
Comment (wenytang-ms) [inline packages/cli/src/utils.ts:207]: do we need consider alpha or beta, not sure gte or lte works fine with version with prerelease postfix.

**Review by jayzhang — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

File: packages/cli/src/utils.ts (hunk 2 above)
Comment (tecton) [reply packages/cli/src/utils.ts:207]: Good point, I tested the `gte()` function with following versions, so I think it should work as expected.
* gte("2.1.0-rc.0", "2.1.0") is false
* gte("2.1.0-rc.0", "2.0.5") is true
* gte("2.1.0-alpha.0", "2.1.0") is false
* gte("2.1.0-alpha.0", "2.0.5") is true

File: packages/cli/src/utils.ts (hunk 1 above)
Comment (tecton) [reply packages/cli/src/utils.ts:204]: It looks like the API name is related to CLI argument name "template-name", I think maybe @jayzhang could help to take a look and decide whether to rename it.

**Review by wenytang-ms — APPROVED** [review]

**Pull request merged**